src/flamenco/gossip/fd_gossip.c

```diff
@@ -1,215 +1,312 @@
+
+//   return 0;
+// }
+
+// static int
+// rx_pull_response( fd_gossip_t *                     gossip,
+//                   fd_gossip_pull_response_t const * pull_response,
+//                   long                              now ) {
+//   /* TODO: use epoch_duration and make timeouts ... ? */
+
+//   for( ulong i=0UL; i<pull_response->values_len; i++ ) {
+//     int upserts = fd_crds_upserts( gossip->crds, pull_response->values[ i ] );
+
+//     if( FD_UNLIKELY( !upserts ) ) {
+//       failed_inserts_append( gossip, pull_response->values[ i ] );
+//       continue;
+//     }
+
+//     /* TODO: Is this jittered in Agave? */
+//     long accept_after_nanos;
+//     if( FD_UNLIKELY( !memcmp( pull_response->sender_pubkey, gossip->identity_pubkey, 32UL ) ) ) {
+//       accept_after_nanos = 0L;
+//     } else if( stake( pull_response->sender_pubkey ) ) {
+//       accept_after_nanos = now-15L*1000L*1000L*1000L;
+//     } else {
+//       accept_after_nanos = now-432000L*1000L*1000L*1000L;
+//     }
+
+//     if( FD_LIKELY( accept_after_nanos<=fd_crds_value_wallclock( pull_response->values[ i ] ) ) ) {
+//       fd_crds_insert( gossip->crds, pull_response->values[ i ], now );
+//       fd_crds_update_record_timestamp( pull_response->sender_pubkey, now );
+//     } else if( fd_crds_has_contact_info( pull_response->sender_pubkey ) ) {
+//       fd_crds_insert( gossip->crds, pull_response->values[ i ], now );
+//     } else {
+//       failed_inserts_append( gossip, pull_response->values[ i ] );
+//     }
+//   }
+
+//   return 0;
+// }
+
+// static int
+// rx_push( fd_gossip_t *            gossip,
+//          fd_gossip_push_t const * push,
+//          long                     now ) {
+//   uchar const * relayer_pubkey = push->sender_pubkey;
+
+//   for( ulong i=0UL; i<push->values_len; i++ ) {
+//     fd_gossip_crds_value_t * value = push->values[ i ];
+//     if( FD_UNLIKELY( value->timestamp<now-30L*1000L*1000L*1000L || value->timestamp>now+30L*1000L*1000L*1000L ) ) continue;
+
+//     uchar const * origin_pubkey = fd_crds_value_pubkey( value );
+
+//     int error = fd_crds_insert( gossip->crds, value, now );
+//     ulong num_duplicates = 0UL;
+//     if( FD_UNLIKELY( error>0 ) )      num_duplicates = (ulong)error;
+//     else if( FD_UNLIKELY( error<0 ) ) num_duplicates = ULONG_MAX;
+
+//     fd_prune_finder_record( gossip->prune_finder, origin_pubkey, relayer_pubkey, num_duplicates );
+//   }
+
+//   return 0;
+// }
 
 static int
 rx_prune( fd_gossip_t *             gossip,
           fd_gossip_prune_t const * prune,
           long                      now ) {
-  if( FD_UNLIKELY( now-500L*1000L*1000L>prune->data->wallclock ) ) return FD_GOSSIP_RX_ERR_PRUNE_TIMEOUT;
-  else if( FD_UNLIKELY( !memcmp( gossip->identity_pubkey, prune->data->destination, 32UL ) ) ) return FD_GOSSIP_RX_ERR_PRUNE_DESTINATION;
+  if( FD_UNLIKELY( now-500L*1000L*1000L>(long)prune->wallclock ) ) return FD_GOSSIP_RX_PRUNE_ERR_TIMEOUT;
```

Comment:
```suggestion
FD_GOSSIP_RX_PRUNE_ERR_STALE;
```
Is probably a little more clear than TIMEOUT